File: template.rs
use utils::Day;

pub fn part_one(input: String) -> Option<i32> {
    None
}

pub fn part_two(input: String) -> Option<i32> {
    None
}

let DAY = Day {
    loc: file!(),
    part_one,
    part_two,
};
pub fn main() {
    let _ = DAY.run();
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_part_one() {
        let input = include_str!("./example.txt");
        assert_eq!(part_one(input), Some(420));
    }

    #[test]
    fn test_part_two() {
        let input = include_str!("./example.txt");
        assert_eq!(part_two(input), Some(420));
    }
}

File: template.py
from utils.execute import Day
from utils.input import Input


def part_one(input: Input) -> int | None:
    pass


def part_two(input: Input) -> int | None:
    pass


if __name__ == "__main__":
    assert (
        p1 := part_one(Input.from_file_relpath("example.txt"))
    ) == None, f"Part one failed, {p1=}"
    assert (
        p2 := part_two(Input.from_file_relpath("example.txt"))
    ) == None, f"Part two failed, {p2=}"
    Day(part_one, part_two).run()
